tmt/steps/provision/testcloud.py

```diff
@@ -213,6 +207,10 @@ def options(cls, how=None):
                 '-c', '--connection',
                 type=click.Choice(['session', 'system']),
                 help="What session type to use, 'session' by default."),
+            click.option(
+                '-a', '--arch',
+                type=click.Choice(['x86_64', 'aarch64', 's390x', 'ppc64le']),
+                help="What architecture to virtualize, host arch by default."),
```

Comment:
Thinking about this, a great start would to decide/agree on architecture naming (eg. ppc64le or just ppc64, aarch64 or arm8/armv8)? I personally don't care much, I am using what python's platform.machine() returns.

And cmdline/class parameter naming. Maybe, ideal would be adding arch to tmt.Guest?

Do you have any other ideas on what could be aligned/decided?

Reply:
Good points. The list of architectures would be a welcome addition, if nothing else, HW requirements would mandate having something like this sooner or later. A recommended or even mandated option name would be good to have, I'd like to avoid two plugins accepting `arch` and others `architecture`. No other ideas at this moment.

Quick note: `ppc64` and `ppc64le` are two distinct architectures, one cannot be used as an alias for the other. In terms of the requested environment, requesting `ppc64le` and getting `ppc64` would be a mistake, the same applies for requesting `ppc64` and getting `ppc64le`.

Reply:
Yes, definitely makes sense to keep the `arch` naming consistent. So far we only have standardized architecture names in the [context dimension](https://tmt.readthedocs.io/en/latest/spec/context.html#dimension) definition. We're using `arch` as the key name there so I'd suggest to go with this. The dimension also explicitly enumerates architectures:

    aarch64, armhfp, i386, ppc64, ppc64le, s390x, x86_64

Those were, I believe, taken from the old test case relevancy implementation to make the context adjustments backward compatible. Unless there are any objections I'd suggest to use the same naming for provision arches as well. I guess we should update the [hardware](https://tmt.readthedocs.io/en/latest/spec/plans.html#hardware) with the arch key as well. I'll submit a PR with the change.

Reply:
The `arch` key proposed in #1148.